File: catkin_ws/src/my_robot_tutorials/src/number_publisher.cpp
#include <ros/ros.h>
#include <std_msgs/Int64.h>
#include <stdlib.h>

int main (int argc, char **argv){
    ros::init(argc, argv, "number_publisher");
    ros::NodeHandle nh;
    
    ros::Publisher pub = nh.advertise<std_msgs::Int64>("/number",10);
    ros::Rate rate(2);

    while(ros::ok()){
        std_msgs::Int64 msg;
        msg.data = rand()%10 +1;
        pub.publish(msg);
        rate.sleep();
    }
}

File: catkin_ws/src/my_robot_tutorials/scripts/number_publisher.py
#!/usr/bin/env python3

import rospy
from std_msgs.msg import Int64
import random

if __name__ == '__main__':
    rospy.init_node("number_publisher")
    pub = rospy.Publisher('/number', Int64, queue_size= 10)
    rate = rospy.Rate(2)

    while not rospy.is_shutdown():
        msg = Int64()
        msg.data = random.randint(1,10) # send random integer
        pub.publish(msg)
        rate.sleep()
    
    rospy.loginfo("Node was stopped")
